### PR TITLE
Adds a table of contents to the readme file.

### PR DESCRIPTION
**Pull request by camilopayan**
I found it really hard to get around the readme to answer my questions quickly. I wasn't familiar enough with it to search for what I needed. To that end, I created a readme file to go with it.

Let me know if there are any changes you want me to make. Great library, btw, I'm really enjoying using it.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -8,6 +8,68 @@ Faker requires PHP >= 5.3.3.
 
 [![Monthly Downloads](https://poser.pugx.org/fzaninotto/faker/d/monthly.png)](https://packagist.org/packages/fzaninotto/faker) [![Build Status](https://secure.travis-ci.org/fzaninotto/Faker.png)](http://travis-ci.org/fzaninotto/Faker) [![SensioLabsInsight](https://insight.sensiolabs.com/projects/eceb78a9-38d4-4ad5-8b6b-b52f323e3549/mini.png)](https://insight.sensiolabs.com/projects/eceb78a9-38d4-4ad5-8b6b-b52f323e3549)
 
+# Table of Contents
+
+- [Installation](#installation)
+- [Basic Usage](#basic-usage)
+- [Formatters](#formatters)
+	- [`FakerProviderBase`](#fakerproviderbase)
```
Comment (fzaninotto) [inline readme.md]: No need to repeat `FakerProvider`, or to make them code formatted. I'd simply label subsections Base, Lorem, Person, etc.

File: readme.md [hunk 2]
```diff
@@ -8,6 +8,68 @@ Faker requires PHP >= 5.3.3.
...
+	- [`FakerProviderMiscellaneous`](#fakerprovidermiscellaneous)
+	- [`FakerProviderBiased`](#fakerproviderbiased)
+- [Unique and Optional modifiers](#unique-and-optional-modifiers)
+- [Localization](#localization)
+- [Populating Entities Using an ORM or an ODM](#populating-entities-using-an-orm-or-an-odm)
+- [Seeding the Generator](#seeding-the-generator)
+- [Faker Internals: Understanding Providers](#faker-internals-understanding-providers)
+- [Real Life Usage](#real-life-usage)
+- [Language specific formatters](#language-specific-formatters)
+	- [`FakerProviderat_ATPayment`](#fakerproviderat_atpayment)
```
Comment (fzaninotto) [inline readme.md]: No need to display the subsections here. It will be impossible to keep up to date anyway.

**Comment by fzaninotto** [comment]
Great idea!

**Comment by camilopayan** [comment]
All right, updated the pull request per your notes. I agree with the notes, btw, wasn't sure if you'd want it code formatted, figured it'd be best to just pass it by you first.

**Comment by fzaninotto** [comment]
Awesome, thanks a lot!

**Pull request merged**